wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java

```diff
@@ -140,6 +145,13 @@ public Connection connect(
 
     final Connection conn = this.connectionProvider.connect(driverProtocol, hostSpec, props);
 
+    if (PropertyDefinition.INVALIDATE_ALL_CONNECTIONS.getBoolean(props)) {
```

Comment:
It should be implemented as a call to PluginService. The only way plugin can communicate to outer world is though PluginService.